storage/txcache/txListForSender.go

```diff
@@ -0,0 +1,193 @@
+package txcache
+
+import (
+	"container/list"
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go/data"
+)
+
+// txListForSender is
+type txListForSender struct {
+	items          *list.List
+	mutex          sync.Mutex
+	copyBatchIndex *list.Element
+	copyBatchSize  int
+	orderNumber    int64
+	sender         string
+}
+
+// txListForSenderNode is a node of the linked list
+type txListForSenderNode struct {
+	txHash []byte
+	tx     data.TransactionHandler
+}
+
+// newTxListForSender creates a new (sorted) list of transactions
+func newTxListForSender(sender string, globalIndex int64) *txListForSender {
+	return &txListForSender{
+		items:       list.New(),
+		orderNumber: globalIndex,
+		sender:      sender,
+	}
+}
+
+// AddTx adds a transaction in sender's list
+// This is a "sorted" insert
+func (listForSender *txListForSender) AddTx(txHash []byte, tx data.TransactionHandler) {
+	// We don't allow concurent interceptor goroutines to mutate a given sender's list
+	listForSender.mutex.Lock()
+
+	nonce := tx.GetNonce()
+	mark := listForSender.findTxWithLargerNonce(nonce)
+	newNode := txListForSenderNode{txHash, tx}
+
+	if mark == nil {
+		listForSender.items.PushBack(newNode)
+	} else {
+		listForSender.items.InsertBefore(newNode, mark)
+	}
+
+	listForSender.mutex.Unlock()
+}
+
+func (listForSender *txListForSender) findTxWithLargerNonce(nonce uint64) *list.Element {
+	for element := listForSender.items.Front(); element != nil; element = element.Next() {
```

Comment:
can you add a comment for this function to only be used under listForSender.mutex.Lock ?